Update kuttl tests to be executed via install_yamls

kajinamit opened the pull request:
This is the prep work to run kuttl tests via install_yamls in CI.

Hard-coded namespace names are removed so that we can run kuttl tests in an arbitrary namespace.

Also, this fixes the environment variable which is named inconsistently.

stuggi commented:
As commented in line, follow ups in plan to
- initialize the conditions, now that https://github.com/openstack-k8s-operators/lib-common/pull/45 is merged
- do a change to lib-common conditon package to be able to use haveSameStateOf Matcher in envtest validation

abays reviewed (approved):
/lgtm

abays commented on controllers/placementapi_controller.go:
```diff
@@ -485,11 +517,23 @@ func (r *PlacementAPIReconciler) reconcileNormal(ctx context.Context, instance *
 		return ctrlResult, nil
 	}
 	instance.Status.ReadyCount = depl.GetDeployment().Status.ReadyReplicas
+	instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
```
I was pondering this same `condition.DeploymentReadyCondition` in some updates to Neutron operator.  In my case, I am only setting this if `instance.IsReady()`.  Just wondering if it might be confusing to set `condition.DeploymentReadyCondition` to `True` without actually checking the deployment's `ReadyReplicas` and/or `instance.IsReady()` (a function which, in most of our operators, usually looks at `ReadyCount` which is set to the deployment's `ReadyReplicas`)?

stuggi replied:
Agreed, I think we should probably do two changes 
1) like you said, set `condition.DeploymentReadyCondition` to True when `ReadyCount is  >= 1` 
2) change `instance.IsReady()` to not just check for ReadyCount, and enhance it to e.g. where services register in keystone, return true for the overall state if `condition.DeploymentReadyCondition` == True AND `condition.KeystoneServiceReadyCondition` == True . Like service is up when successfully registered in keystone + have a pod up to serve it

gibizer commented:
/lgtm
thank you for updating the envtest too

kajinamit committed:
Update kuttl tests to run in arbitrary namespace

This removes the hard-coded namespace names so that we can run kuttl
tests in an arbitrary namespace.

kajinamit committed:
Rename environment variable to point kuttl test directory

... to make the naming consistent with the other operators.

kajinamit committed:
Do not assert controller-manager deployment

kajinamit committed:
Fix type mismatch in kuttl tests

kajinamit committed:
Fix endpoint assertion in arbitrary namespaces

kajinamit committed:
Rename test directory

... to make the name consistent with the other operators.

kajinamit commented:
/test precommit-check

gibizer reviewed (commented):
Together with https://github.com/openstack-k8s-operators/install_yamls/pull/383 I was able to run the placement_kuttl test locally successfully
```
$ PATH="/home/gibi/bin/sh-to-bash:$PATH" PLACEMENT_KUTTL_CONF=../placement-operator/kuttl-test.yaml  PLACEMENT_KUTTL_DIR=../placement-operator/tests/kuttl/tests make placement_kuttl

[snip]

=== CONT  kuttl
    harness.go:399: run tests finished
    harness.go:508: cleaning up
    harness.go:563: removing temp folder: ""
--- PASS: kuttl (81.43s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/placement_scale (73.62s)
```

So this looks good to me.

I'd like to get @jamepark4 to take a look too.

abays reviewed (approved):
/lgtm
